sdk/storage/AzureStorageBlob/Source/Data/BlobTransfer+CoreDataClass.swift

```diff
@@ -107,6 +107,7 @@ extension BlobTransfer {
         transfer.endRange = endRange
         transfer.state = .pending
         transfer.transferType = type
+        transfer.id = UUID()
```

Comment:
Might be a six in one hand, half dozen in the other kind of thing, but any particular reason you choose to make id UUID across the board instead of base64 encoded UUID? Since the service needs the string version, it seems we could avoid potential conversion bug by just making them all base64 strings and not needed to worry about when/if to convert them.

Reply:
I think it was because I started here and only later understood why we were converting it to a base64 string. As a developer if I receive an item from an ORM I would expect it to have an ID that's an integer or a UUID. The fact that the service uses a base64-encoded string in its API spec is an implementation detail of the service I didn't feel we should leak. So I kept the conversion being performed just-in-time right before we actually communicate with the service.